### PR TITLE
fix "pristine" migration

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

File: card/mod/core/set/all/states.rb [hunk 1]
```diff
@@ -76,7 +76,9 @@ def new?
 
 # has not been edited directly by human users.  bleep blorp.
 def pristine?
-  new_card? || !actions.joins(:act).where(
-    "card_acts.actor_id != ?", Card::WagnBotID
-  ).exists?
+  new_card? || !user_changes?
```
Comment (ethn) [inline card/mod/core/set/all/states.rb]: so much nicer to read.

File: card/lib/card/migration.rb [hunk 2]
```diff
@@ -102,6 +102,13 @@ def merge_cards names_or_keys
       Card::Migration::Import.new(data_path).merge only: names_or_keys
     end
 
+    def merge_pristine_cards names_or_keys
+      names_or_keys = Array(names_or_keys)
+
+      pristine = names_or_keys.select { |n| !Card.exists?(n) || Card.fetch(n)&.pristine? }
```
Comment (ethn) [inline card/lib/card/migration.rb]: isn't the `!Card.exists?` unnecessary  here?

Comment (ethn) [reply card/lib/card/migration.rb]: Oh, I guess you'd need to fetch with "new", eg:
`Card.fetch(n, new: {}).pristine?`

Comment (xithan) [reply card/lib/card/migration.rb]: Yes, I noticed that that would work while I was fixing this but it didn't feel like an improvement.
(the real issue was the missing ! on the select call)

**Pull request merged**